### PR TITLE
Added Poison Effect Upon leaving room's area

### PR DESCRIPTION
**Pull request by RealMangorage**
Added the Poison affect so that when players leave their room by somehow getting out of it and into the void of Compact Dimension itll do the following:

Give Them poison and Blindness
Teleports em back to the machine they last went into (if none then it just continues to poison them)

I did it based on what it does in 1.12

### CODE REVIEW

**Comment by Apceniy** [comment]
But in 1.12 it teleports you out of the machine, not back into it

File: src/main/java/dev/compactmods/machines/wall/MachineVoidAir.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import net.minecraft.world.effect.MobEffects;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.AirBlock;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.Material;
+
+public class MachineVoidAir extends AirBlock {
+    final public static DamageSource DAMAGE_SOURCE = new DamageSource("machinevoidair");
```
Comment (robotgryphon) [inline src/main/java/dev/compactmods/machines/wall/MachineVoidAir.java]: Consider underscores/prefix of mod id here

File: src/main/java/dev/compactmods/machines/wall/MachineVoidAir.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    public void entityInside(BlockState pState, Level pLevel, BlockPos pPos, Entity pEntity) {
+        if (pLevel.isClientSide) return;
+
+        if (pEntity instanceof ServerPlayer player) {
+            if (player.isCreative()) return;
+
+            if (player.getActiveEffectsMap().containsKey(MobEffects.BLINDNESS))
+                PlayerUtil.teleportPlayerBackToMachine((ServerLevel) pLevel, player);
```
Comment (robotgryphon) [inline src/main/java/dev/compactmods/machines/wall/MachineVoidAir.java]: 1.12 teleported the player fully out of the machine, prefer that util method

File: src/main/java/dev/compactmods/machines/upgrade/ChunkloadUpgradeItem.java [hunk 3]
```diff
@@ -10,16 +11,16 @@ public class ChunkloadUpgradeItem extends RoomUpgradeItem {
     public ChunkloadUpgradeItem(Properties props) {
         super(props);
     }
-
     @Override
     public void fillItemCategory(CreativeModeTab tab, NonNullList<ItemStack> stacks) {
+        if (tab != CompactMachines.COMPACT_MACHINES_ITEMS) return;
```
Comment (robotgryphon) [inline src/main/java/dev/compactmods/machines/upgrade/ChunkloadUpgradeItem.java]: Unrelated, please revert/open a new PR

File: src/datagen/java/dev/compactmods/machines/datagen/lang/EnglishLangGenerator.java [hunk 4]
```diff
@@ -120,6 +121,7 @@ protected void addTranslations() {
         add("jei.compactmachines.machines", "Machines are used to make pocket dimensions. Craft a machine and place it in world, then use a Personal Shrinking Device to go inside.");
         add("jei.compactmachines.shrinking_device", "Use the Personal Shrinking Device (PSD) on a machine in order to enter a compact space. " +
                 "You can also right click it in the overworld for more info.");
+        add("death.attack." + MachineVoidAir.DAMAGE_SOURCE.msgId, "%1$s Died from Machine Void Air");
```
Comment (robotgryphon) [inline src/datagen/java/dev/compactmods/machines/datagen/lang/EnglishLangGenerator.java]: Would prefer to align with other death messages, ie "`player` failed to enter the void"

**Review by robotgryphon — CHANGES_REQUESTED** [review]

**Comment by RealMangorage** [comment]
Ok I did I fixed em.

File: src/main/java/dev/compactmods/machines/config/ServerConfig.java [hunk 5]
```diff
@@ -76,6 +78,10 @@ private static void generateConfig() {
                 .comment("Command level required for changing room upgrades.")
                 .defineInRange("upgrades", Commands.LEVEL_GAMEMASTERS, Commands.LEVEL_ALL, Commands.LEVEL_OWNERS);
 
+        ALLOWED_OUTSIDE_MACHINE = builder
+                .comment("If true it'll teleport player out of machine dimension if player is outside of the machine (default: false)")
+                .define("allowoutside", false);
+
```
Comment (robotgryphon) [inline src/main/java/dev/compactmods/machines/config/ServerConfig.java]: Needs moved to a new section, this is in the `commands` section. Also, prefer Java casing, `allowOutside`

File: src/main/java/dev/compactmods/machines/util/PlayerUtil.java [hunk 6]
```diff
@@ -17,11 +18,13 @@
 import dev.compactmods.machines.room.exceptions.NonexistentRoomException;
 import dev.compactmods.machines.room.history.PlayerRoomHistoryItem;
 import net.minecraft.core.BlockPos;
+import net.minecraft.network.chat.Component;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.GameType;
```
Comment (robotgryphon) [inline src/main/java/dev/compactmods/machines/util/PlayerUtil.java]: Clean up imports

File: src/main/java/dev/compactmods/machines/util/PlayerUtil.java [hunk 7]
```diff
@@ -147,6 +150,38 @@ public static void teleportPlayerOutOfMachine(ServerLevel world, @Nonnull Server
...
+                            teleportPlayerIntoRoom(serv, serverPlayer, room, false);
+                        } catch (MissingDimensionException | NonexistentRoomException e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                }
+            }
+        });
+    }
+
```
Comment (robotgryphon) [inline src/main/java/dev/compactmods/machines/util/PlayerUtil.java]: Evaluating if this is necessary; if the config option isn't an option list between tele back and tele out, this should probably just be removed. If kept, the non-existent history conditional block should fall into returning the player back to their spawn (`teleportPlayerToRespawnOrOverworld`)

File: src/main/java/dev/compactmods/machines/upgrade/RoomUpgradeItem.java [hunk 8]
```diff
@@ -24,6 +24,8 @@ public RoomUpgradeItem(Properties props) {
 
     public abstract RoomUpgrade getUpgradeType();
 
+
+
```
Comment (robotgryphon) [inline src/main/java/dev/compactmods/machines/upgrade/RoomUpgradeItem.java]: Remove rogue spacing please

File: src/main/java/dev/compactmods/machines/upgrade/ChunkloadUpgradeItem.java [hunk 9]
```diff
@@ -1,5 +1,6 @@
 package dev.compactmods.machines.upgrade;
 
+import dev.compactmods.machines.CompactMachines;
```
Comment (robotgryphon) [inline src/main/java/dev/compactmods/machines/upgrade/ChunkloadUpgradeItem.java]: Remove unnecessary file changes, to reduce patch size

**Review by robotgryphon — CHANGES_REQUESTED** [review]
Looking much better; one more cleanup pass and it should be clear

**Review by robotgryphon — APPROVED** [review]

**Pull request merged**